src/widgets/query-rule-custom-data/query-rule-custom-data.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { render, unmountComponentAtNode } from 'preact-compat';
+import {
+  getContainerNode,
+  prepareTemplateProps,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { WidgetFactory } from '../../types';
+import connectQueryRules, {
+  QueryRulesRenderer,
+} from '../../connectors/query-rules/connectQueryRules';
+import CustomData from '../../components/QueryRuleCustomData/QueryRuleCustomData';
+
+export type QueryRuleCustomDataTemplates = {
+  default: string | ((items: object[]) => string);
+};
+
+type QueryRuleCustomDataWidgetParams = {
+  container: string | HTMLElement;
+  templates?: QueryRuleCustomDataTemplates;
+  transformItems?: (items: object[]) => any;
+};
+
+interface QueryRuleCustomDataConnectorWidgetParams
```

Comment:
Why this type is useful? We can directly provide `QueryRuleCustomDataWidgetParams` to the renderer.

Reply:
Nope, because at the renderer level `container` _must_ be an `HTMLElement`, and not `string | HTMLElement`.

Reply:
Yes indeed, a different name would have helped to understand IMO. The `container` has nothing to do with the connector it's only used for the rendering. It's confusing to see it in the name.

Reply:
I renamed it `QueryRuleCustomDataRendererWidgetParams`.